lib/core/src/wallet.rs

```diff
@@ -224,38 +228,16 @@ impl OnchainWallet for LiquidOnchainWallet {
         Ok(())
     }
 
-    fn derive_bip32_key(&self, path: Vec<ChildNumber>) -> Result<ExtendedPrivKey, PaymentError> {
-        let seed = self.lwk_signer.seed().ok_or(PaymentError::SignerError {
-            err: "Could not get signer seed".to_string(),
-        })?;
-
-        let bip32_xpriv = ExtendedPrivKey::new_master(self.config.network.into(), &seed)?
-            .derive_priv(&Secp256k1::new(), &path)?;
-        Ok(bip32_xpriv)
-    }
-
     fn sign_message(&self, message: &str) -> Result<String> {
-        let seed = self
-            .lwk_signer
-            .seed()
-            .ok_or(anyhow!("Could not get signer seed"))?;
-        let secp = Secp256k1::new();
-        let keypair = ExtendedPrivKey::new_master(self.config.network.into(), &seed)
-            .map_err(|e| anyhow!("Could not get signer keypair: {e}"))?
-            .to_keypair(&secp);
         // Prefix and double hash message
         let mut engine = sha256::HashEngine::default();
         engine.write_all(LN_MESSAGE_PREFIX)?;
         engine.write_all(message.as_bytes())?;
         let hashed_msg = sha256::Hash::from_engine(engine);
-        let double_hashed_msg = Message::from(sha256::Hash::hash(&hashed_msg));
+        let double_hashed_msg = Message::from_digest(hashed_msg.into_inner());
```

Comment:
This doesn't seem to be double hashing any more?

Reply:
Indeed calling `sign-message` / `check-message` with the CLI finds the signature as invalid.

Changing the line to
```rust
let double_hashed_msg = Message::from_digest(sha256::Hash::hash(&hashed_msg).into_inner());
```
fixes the problem.

Reply:
Good catch guys. I fixed that and added a test for this case.